### PR TITLE
Start using nuget packages built by dotnet CLI 

### PR DESCRIPTION
**Pull request by jtattermusch**
Based on #8075.

### CODE REVIEW

File: src/csharp/build_packages.bat [hunk 1]
```diff
@@ -77,8 +81,8 @@ endlocal
 xcopy /Y /I *.nupkg ..\..\artifacts\
 
 @rem create a zipfile with the artifacts as well
-powershell -Command "Add-Type -Assembly 'System.IO.Compression.FileSystem'; [System.IO.Compression.ZipFile]::CreateFromDirectory('..\..\artifacts', 'csharp_nugets.zip');"
-xcopy /Y /I csharp_nugets.zip ..\..\artifacts\
+powershell -Command "Add-Type -Assembly 'System.IO.Compression.FileSystem'; [System.IO.Compression.ZipFile]::CreateFromDirectory('..\..\artifacts', 'csharp_nugets_obsolete.zip');"
+xcopy /Y /I csharp_nugets_obsolete.zip ..\..\artifacts\
```
Comment (apolcyn) [inline src/csharp/build_packages.bat]: would we want to get rid of this package script?

Comment (jtattermusch) [inline src/csharp/build_packages.bat]: at some point, yes. But I'd prefer to do that once the new nugets are published and well tested.

**Comment by jtattermusch** [comment]
Seems that there are still some issues with building the packages.
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_packages/1722/platform=linux/console

**Comment by jtattermusch** [comment]
Distribtest run to watch:
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/1674/

**Comment by apolcyn** [comment]
I think the linux distrib test failure on linux is just the issue with centos and nuget https://github.com/grpc/grpc/issues/8070. 

Seems issues with finding ..\packages\Grpc.Core.1.1.0-dev\build\net45\Grpc.Core.target on windows though.

**Comment by jtattermusch** [comment]
Ok, so problems on Windows and Mac are once again with nuget.

It looks like the nuget version on our mac workers is 3.X (nuget restore doesn't work with mono):
``
nuget restore
MSBuild auto-detection: using msbuild version '4.0' from '/Library/Frameworks/Mono.framework/Versions/4.2.2/lib/mono/4.5'.
MsBuild.exe does not exist at '/Library/Frameworks/Mono.framework/Versions/4.2.2/lib/mono/4.5/msbuild.exe'.
```

On windows, the  "..\packages\Grpc.Core.1.1.0-dev\build\net45\Grpc.Core.target" missing also means that something went wrong with nuget restore (i checked and the file is there).

**Comment by jtattermusch** [comment]
After the last commit, all distribtests are green with the exception centos7 (which should be addressed separately).

**Comment by jtattermusch** [comment]
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/1681/

**Comment by apolcyn** [comment]
LGTM

**Comment by jtattermusch** [comment]
All required statuses are green and yet github still doesn't allow me to merge ("Required statuses must pass before merging"). Can we please override?

**Comment by nicolasnoble** [comment]
Something seems broken in github...

**Pull request merged**